assets/shaders/array_texture.wgsl

```diff
@@ -0,0 +1,63 @@
+#import bevy_pbr::mesh_view_bindings
+#import bevy_pbr::mesh_bindings
+
+#import bevy_pbr::pbr_types
+#import bevy_pbr::utils
+#import bevy_pbr::clustered_forward
+#import bevy_pbr::lighting
+#import bevy_pbr::shadows
+#import bevy_pbr::pbr_functions
+
+[[group(1), binding(0)]]
+var my_array_texture: texture_2d_array<f32>;
+[[group(1), binding(1)]]
+var my_array_texture_sampler: sampler;
+
+struct FragmentInput {
+    [[builtin(front_facing)]] is_front: bool;
+    [[builtin(position)]] frag_coord: vec4<f32>;
+    [[location(0)]] world_position: vec4<f32>;
+    [[location(1)]] world_normal: vec3<f32>;
+    [[location(2)]] uv: vec2<f32>;
+#ifdef VERTEX_TANGENTS
+    [[location(3)]] world_tangent: vec4<f32>;
+#endif
+#ifdef VERTEX_COLORS
+    [[location(4)]] color: vec4<f32>;
+#endif
+};
+
+[[stage(fragment)]]
+fn fragment(in: FragmentInput) -> [[location(0)]] vec4<f32> {
+    let layer = i32(in.world_position.x) & 0x3;
+
+    // Prepare a 'processed' StandardMaterial by sampling all textures to resolve
+    // the material members
+    var pbr_input: PbrInput = pbr_input_new();
```

Comment:
Random thought (not necessary for this pr): maybe we should have a pbr_input_default() function that sets as much as possible "automatically". Hard to do for FragmentInput, given that it is defined per-shader. But we could at least auto-calculate is_orthographic. Maybe we could have a standardized struct for "normal" fragment inputs, which users pass in to pbr_input? Again, definitely not something to do in this pr, but it feels like theres room for ux improvements here.